### PR TITLE
feat(bundle): add support for --reload flag

### PR DESCRIPTION
**Pull request by JayHelton**
###Description
Adding support for the reload flag for the bundle command

- [x] Add to args parse for bundle
- [x] Manual Testing
- [x] Fix Any Broken Tests
- [x] Add Tests

###References
#6973 

**Current Observations:**

> It looks like the reload flag is based onto global state, and then the file `global.file_fetcher`.
> This would imply that denos global state will handle this scenario simply by providing the flag to the init of state. Testing will confirm.

This was proven true.

### CODE REVIEW

**Comment by JayHelton** [comment]
Some screenshots of the added functionality

Bundle after manual deletion of a single cached dep
Screen Shot 2020-08-09 at 10 00 14 PM

Bundle with all cached items, no reload
Screen Shot 2020-08-09 at 9 53 50 PM

Bundle with all reload
Screen Shot 2020-08-09 at 9 53 29 PM

Bundle with single reload
Screen Shot 2020-08-09 at 9 53 11 PM

**Comment by JayHelton** [comment]
@lucacasonato This should be functionally good to go.

I do have some light concerns around the unit tests around the bundle command parsers.

Nothing I touched seemed to have tests explicitly for the functions.

Is there a preference for where I would add a test for the logic I added? It seems like there may be a higher level test module I might be able to add a test case to.

File: cli/flags.rs [hunk 1]
```diff
@@ -761,6 +762,7 @@ fn bundle_subcommand<'a, 'b>() -> App<'a, 'b> {
     )
     .arg(Arg::with_name("out_file").takes_value(true).required(false))
     .arg(ca_file_arg())
+    .arg(reload_arg())
     .arg(importmap_arg())
     .arg(unstable_arg())
```
Comment (bartlomieju) [inline cli/flags.rs:767]: Please add a unit test similar to `bundle_with_lock` that tests `--reload` flag is accepted. Otherwise LGTM

Comment (JayHelton) [reply cli/flags.rs:767]: Beautiful! Thanks for pointing me in the right direction

**Pull request merged**